### PR TITLE
Add redirect behavior

### PR DESCRIPTION
**Pull request by sweatC**
we need to redirect all requests to http://jetthoughts.com/blog/* to https://jtway.co
image

- Add jekyll-redirect-from gem file.
- Configure gem package.
- Add blog.html redirect.

UPD: Move old post to the Medium. If we don't have this posts on Medium then setup redirect.
image

### CODE REVIEW

**Comment by sweatC** [comment]
https://www.useloom.com/share/b4ef352c60a74a0692ab852c6137044a

File: redirects/blog.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+permalink: /blog
+redirect_to:
+  - https://jtway.co/
+---
```
Comment (andriyParashchuk) [inline redirects/blog.html]: Add new line

**Comment by pftg** [comment]
@sweatC any status for this issue?

**Comment by pftg** [comment]
@sweatC please deploy

**Comment by sweatC** [comment]
@pftg deployed: http://jt-pr290.surge.sh/

**Comment by obliviusm** [comment]
doesn't work on surge
@sweatC is testing on GH pages

File: blog.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+redirect_from:
+  - /blog/tech
```
Comment (obliviusm) [inline blog.html:3]: is it possible to redirect with wild card `/blog/*` => `https://jtway.co/`

Comment (pftg) [reply blog.html:3]: @obliviusm this plugin does not allow such redirects

**Comment by pftg** [comment]
@sweatC I see not found for list of pages: https://cl.ly/1H3p2S043P2N/Image%202018-05-02%20at%2009.26.06.public.png

but expected redirect

**Comment by sweatC** [comment]
@pftg @obliviusm @andriyParashchuk pls check: http://jt-pr290.surge.sh/
Redirect works fine, but we need to change url for gh pages production. For now it's set to staging.
https://www.useloom.com/share/2e42c0d8c16145f098412220cdeb6aea

File: blog.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  - /blog/tricks
+  - /blog/tricks/2012/11/28/multiple-likes-for-one-page.html
+  - /blog/tech/2011/09/24/change-factory-girl-directory-path-for-rails-3-generator.html
+  - /blog/tech/2010/07/11/geoipcity-on-mac-os.html
+  - /blog/tricks/2009/12/16/postgresql-insert-not-valid-unicode-values.html
+  - /blog/tech/2010/09/26/installing-pg-gem-on-mac-os.html
```
Comment (pftg) [inline blog.html]: I think better to copy all posts from the old blog and left only redirects without any content, and for guys who without js say that blog moved to and new url

**Review by pftg — CHANGES_REQUESTED** [review]
Let's finish all at once

**Comment by pftg** [comment]
@sweatC please move old posts to new blog, update their body to show that we has been moved to jtway.co and new link

File: bin/production [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+chdir APP_ROOT do
+  # This script is building production version of your application.
+  # It uses livereload, so you don't need to stop it manually.
+  puts '== Building Production =='
+  system!('JEKYLL_ENV=production bundle exec jekyll build --watch')
+end
```
Comment (pftg) [inline bin/production]: 1. this is a build, not production cli.
2. no need to use ruby for scripts, will be enough to have bash https://github.com/jetthoughts/jetthoughts.github.io/blob/develop/bin/staging-release

File: bin/deploy [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+chdir APP_ROOT do
+  domain = ARGV[0] # first argument from the command line
+
+  # This script is deploying your application on Surge domain.
+  # You just need to put first argument, it is name of your Surge domain.
+  puts "== Deploying on #{domain} =="
+  system!("surge -p ./_site -d #{domain}")
+end
```
Comment (pftg) [inline bin/deploy]: 1. `bin/release`
2. add build in the script in order to recompile and deploy in one command
3. please use bash

File: _config.yml [hunk 6]
```diff
@@ -14,12 +14,13 @@
 # You can create any custom variable you would like, and they will be accessible
 # in the templates via {{ site.myvariable }}.
 name: JetThoughts
-url: http://www.jetthoughts.com
+url: http://jt-pr290.surge.sh
```
Comment (andriyParashchuk) [inline _config.yml]: You do not need to save the staging url in order to deploy to surge.sh

Comment (sweatC) [reply _config.yml]: I do this for another purpose. For redirects.

Comment (andriyParashchuk) [reply _config.yml]: why this url in the PR?
you deploy to surge.sh and forgot the url

File: bin/production (hunk 4 above)
Comment (sweatC) [reply bin/production]: Accepted

File: bin/deploy (hunk 5 above)
Comment (sweatC) [reply bin/deploy]: Accepted

File: _config.yml (hunk 6 above)
Comment (sweatC) [reply _config.yml]: It is only for testing redirects gem.

**Comment by sweatC** [comment]
@pftg without JS it redirects, and shows document with redirection link, if redirection isn't completed.
https://www.useloom.com/share/65bbcfaa9c5640c784b25b6f970d56af

**Comment by sweatC** [comment]
Status: Tomorrow will work on adding more articles from archive.

**Comment by pftg** [comment]
@sweatC what is the status for this issue?

**Comment by sweatC** [comment]
@pftg it is ready for production

**Comment by sweatC** [comment]
@pftg pls check

**Comment by pftg** [comment]
@sweatC please deploy on surge.sh

**Comment by sweatC** [comment]
@pftg before merging with master I need to change _config.url and rebase all commits.
Staging for verify: http://jt-pr290.surge.sh

**Comment by pftg** [comment]
@sweatC I tried this https://cl.ly/3w05133e0T1x/Image%202018-05-08%20at%2014.37.21.public.png but have not found any redirects. Could you record gif is it working or not your side

**Comment by pftg** [comment]
@sweatC looks like is working

**Comment by sweatC** [comment]
@pftg pls check
https://www.useloom.com/share/41121dbc450e438bb8ab2cbbd70638e5

**Comment by pftg** [comment]
@sweatC please squash and reabse

**Review by pftg — APPROVED** [review]
LGTM

**Comment by sweatC** [comment]
@pftg please merge

**Comment by obliviusm** [comment]
@sweatC please do rebase instead of merge

**Pull request merged**